src/sentry/integrations/pipeline.py

```diff
@@ -50,10 +51,27 @@ class IntegrationPipeline(Pipeline):
 
     def finish_pipeline(self):
         data = self.provider.build_integration(self.state.data)
+        try:
+            response = self.reinstall(data)
+        except Integration.DoesNotExist:
+            pass
+        else:
+            return response
         response = self._finish_pipeline(data)
         self.clear_session()
         return response
 
+    def reinstall(self, data):
```

Comment:
Do we actually need a reinstall method? The _finish_pipeline should be idempotent enough that running through it will be the equivalent of a reinstall.

We would need to set the `status` of the integration and then call the reinstall stuff if the organization integration wasn't created during the install